source/math/FloatVectorList.ooc

```diff
@@ -381,4 +382,68 @@ FloatVectorList: class extends VectorList<Float> {
 			result add(this[i] clamp(floor, ceiling))
 		result
 	}
+	findMaxIndex: func -> Int {
+		index := 0
+		thisPointer := (this pointer as Float*)
+		for (i in 1 .. this count)
+			if (thisPointer[index] < thisPointer[i])
+				index = i
+		index
+	}
+	crossCorrelation: func (other: This, vectorLength: Int) -> Float {
```

Comment:
Correlation is a measure of similarity between two data sets and is a scalar. Cross-correlation is a set of correlations where one of the data sets is shifted with different lags.